Runtime/ui/txt/layout.cs

```diff
@@ -1,218 +1,284 @@
-                    float letterSpace = style.letterSpacing;
-                    float letterSpaceHalfLeft = letterSpace * 0.5f;
-                    float letterSpaceHalfRight = letterSpace - letterSpaceHalfLeft;
-
-                    x += letterSpaceHalfLeft;
-                    this._advances[i] += letterSpaceHalfLeft;
-
-                    var metrics = FontMetrics.fromFont(font, style.UnityFontSize);
-
-                    var minX = x;
-                    var maxX = metrics.descent - metrics.ascent + x;
-                    var minY = metrics.ascent;
-                    var maxY = metrics.descent;
-
-                    if (this._bounds.width <= 0 || this._bounds.height <= 0) {
-                        this._bounds = UnityEngine.Rect.MinMaxRect(
-                            minX, minY, maxX, maxY);
-                    }
-                    else {
-                        if (minX < this._bounds.x) {
-                            this._bounds.x = minX;
-                        }
-
-                        if (minY < this._bounds.y) {
-                            this._bounds.y = minY;
-                        }
-
-                        if (maxX > this._bounds.xMax) {
-                            this._bounds.xMax = maxX;
-                        }
-
-                        if (maxY > this._bounds.yMax) {
-                            this._bounds.yMax = maxY;
-                        }
-                    }
-
-                    this._positions[i] = x;
-                    float advance = style.fontSize;
-                    x += advance;
-
-                    this._advances[i] += advance;
-                    this._advances[i] += letterSpaceHalfRight;
-                    x += letterSpaceHalfRight;
-                }
-                else {
-                    this._advances[i] = 0;
-                    this._positions[i] = x;
-                }
-
-                this._advance = x;
-            }
-        }
-
-        public void setTabStops(TabStops tabStops) {
-            this._tabStops = tabStops;
-        }
-
-        public int nGlyphs() {
-            return this._count;
-        }
-
-        public List<float> getAdvances() {
-            return this._advances;
-        }
-
-        public float getAdvance() {
-            return this._advance;
-        }
-
-        public float getX(int index) {
-            return this._positions[index];
-        }
-
-        public float getY(int index) {
-            return 0;
-        }
-
-        public float getCharAdvance(int index) {
-            return this._advances[index];
-        }
-
-        public Rect getBounds() {
-            return Rect.fromLTWH(this._bounds.x, this._bounds.y, this._bounds.width, this._bounds.height);
-        }
-    }
+using UnityEngine;
+
+namespace Unity.UIWidgets.ui {
+    static class Layout {
+        // Measure the length of the span of the text. Currently, this is only used to compute the length

```

Comment:
is this logic simplified in this PR or the original codes are already simplified?

we should make our logic the same as in the original codes

Reply:
If you mean compared to Flutter Engine, the logic in the original code here is way simpler than in Flutter Engine. Other simplifications are hacks made in this PR, to save memory and time.
If you mean the structure of the code should be kept, I think I can take some time to make sure that.